src/components/UI/EditableValueLabel.tsx

```diff
@@ -0,0 +1,199 @@
+import styled from '@emotion/styled';
+import PropTypes from 'prop-types';
+import React, {
+  KeyboardEventHandler,
+  Reducer,
+  useReducer,
+  useRef,
+  useState,
+} from 'react';
+import Overlay from 'react-bootstrap/lib/Overlay';
+import Tooltip from 'react-bootstrap/lib/Tooltip';
+import Button from 'UI/Button';
+import { InputElement } from 'UI/Input';
+import { IValidation, IValidationFunction } from 'utils/labelUtils';
+
+interface IEditableValueLabel {
+  allowEdit: boolean;
+  isNew: boolean;
+  onCancel(): void;
+  onEdit(): void;
+  onSave(value: string): void;
+  validationFunc: IValidationFunction;
+  value: string;
+}
+
+interface IReducerState extends IValidation {
+  internalValue: string;
+}
+
+const EditModeWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  margin: 10px 0 10px 5px;
+`;
+
+const ValueInput = styled(InputElement)`
+  margin-right: 10px;
+`;
+
+const EditValueTooltip = styled(Tooltip)`
+  &.in {
+    opacity: 1;
+  }
+  .tooltip-inner {
+    background-color: ${({ theme }) => theme.colors.darkBlueLighter3};
+  }
+  &.top .tooltip-arrow {
+    border-top-color: ${({ theme }) => theme.colors.darkBlueLighter3};
+  }
+`;
+
+const ValidationErrorTooltip = styled(Tooltip)`
+  &.in {
+    opacity: 1;
+  }
+  .tooltip-inner {
+    background-color: ${({ theme }) => theme.colors.error};
+    color: black;
+  }
+  &.top .tooltip-arrow {
+    border-top-color: ${({ theme }) => theme.colors.error};
+  }
+`;
+
+const Editable = styled.span<{ allowEdit: boolean }>`
+  text-decoration: ${({ allowEdit }) => (allowEdit ? 'underline' : 'none')};
+  text-decoration-style: dotted;
+  cursor: ${({ allowEdit }) => (allowEdit ? 'pointer' : 'default')};
+  opacity: ${({ allowEdit }) => (allowEdit ? '1' : '0.6')};
+`;
+
+const EditableValueLabel = ({
+  allowEdit,
+  isNew,
+  onCancel,
+  onEdit,
+  onSave,
+  validationFunc,
+  value,
+}: IEditableValueLabel) => {
```

Comment:
You can use the same `React.FC` generic here for defining the component, as you used in other components

Reply:
Yeah I've tried, but this didn't go well with the PropTypes.

For example with the `allowEdit` prop:
```
Type 'Requireable<boolean>' is not assignable to type 'Validator<boolean>'.
  Types of property '[nominalTypeHack]' are incompatible.
    Type '{ type: boolean | null | undefined; } | undefined' is not assignable to type '{ type: boolean; } | undefined'.
      Type '{ type: boolean | null | undefined; }' is not assignable to type '{ type: boolean; }'.
        Types of property 'type' are incompatible.
          Type 'boolean | null | undefined' is not assignable to type 'boolean'.
            Type 'undefined' is not assignable to type 'boolean'.ts(2322)
```

I think this is related to this issue https://github.com/DefinitelyTyped/DefinitelyTyped/issues/28015 but couldn't figure this out.

Reply:
That error is probably because these props are defined as optional in PropTypes, but as mandatory in the props interface.
```
  allowEdit: PropTypes.bool,
  isNew: PropTypes.bool,
  onCancel: PropTypes.func,
  onEdit: PropTypes.func,
  onSave: PropTypes.func,
  validationFunc: PropTypes.func,
```

In the interface:

```
  allowEdit: boolean;
  isNew: boolean;
  onCancel(): void;
  onEdit(): void;
  onSave(value: string): void;
  validationFunc: IValidationFunction;
```

In order for that to work, you have to either make the props required in PropTypes, or make them optional in the props interface